Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -657,6 +661,35 @@ public void run() {
     });
   }
 
+  @NotNull
+  public List<HoverInformation> analysis_getHover(@NotNull final String _filePath, final int offset) {
+    final String filePath = FileUtil.toSystemDependentName(_filePath);
+    final AtomicReference<List<HoverInformation>> result = new AtomicReference<List<HoverInformation>>();
```

Comment:
You do not use 'atomic' features of this reference. That's correct, we do not need them, we are not afraid of a race here. So simple `Ref` that we use in all other methods here is enough.